doc/api.md

```diff
@@ -0,0 +1,1326 @@
+        "geom": {
+            "srid": 4326,
+            "x": -112.09479999999999,
+            "y": 33.5162
+        },
+        "mapfeature_ptr": 4,
+        "udfs": {
+        },
+        "instance": 2,
+        "updated_at": "2020-07-27T20:22:32.058Z",
+        "readonly": false,
+        "hide_at_zoom": null,
+        "owner_orig_id": null,
+        "width": null,
+        "updated_by": 1,
+        "address_city": null
+    },
+    "share": {
+        "title": "Missing Species on Demo Instance",
+        "description": "This Missing Species is mapped on Demo Intance",
+        "image": "https://www.opentreemap.org/static/img/tree.png",
+        "url": "https://www.opentreemap.org/demo/features/4/"
+    },
+    "progress_messages": [
+        "Add the species",
+        "Add a photo"
+    ],
+    "tree": {
+        "species": null,
+        "height": 42.0,
+        "date_removed": null,
+        "udf:Stewardship": [],
+        "udfs": {
+        },
+        "instance": 2,
+        "id": 1,
+        "canopy_height": null,
+        "readonly": false,
+        "date_planted": null,
+        "diameter": 12.0,
+        "plot": 4
+    },
+    "title": "Missing Species",
+    "feature_type": "Plot",
+    "recent_activity": [
+        [
+            {
+                "id": 1,
+                "username": "demouser"
+            },
+            "2020-07-27T20:22:32.086Z",
+            [
+                {
+                    "ref": null,
+                    "model": "Tree",
+                    "current_value": "42.0",
+                    "field": "height",
+                    "instance_id": 2,
+                    "created": "2020-07-27 20:22:32.086498+00:00",
+                    "user_id": 1,
+                    "requires_auth": false,
+                    "previous_value": null,
+                    "action": 1,
+                    "model_id": 1
+                },
+            ]
+        ]
+    ],
+    "favorited": false
+}
+```
+
+---
+
+`POST /instance/{instance-url-name}/plots/{plot-id}/tree/photo`
+
+Upload a photo of the current tree associated with the specified plot.
+
+The request body should be an image file in a format supported by [Pillow](https://pypi.org/project/Pillow/) 
+
+Returns a JSON object representing the photo.
+
+### HMAC
+
+All OTM API requests must be signed using
+[HMAC](https://enwikipediaorg/wiki/HMAC). The server-side HMAC implementation is based on the [`hmac` package included with Python](https://docs.python.org/3/library/hmac.html)
+
+The [`hmaccurl.py`](../scripts/hmaccurl.py) command line tool is an example of
+how API requests can are signed with HMAC. It is a Python script that wraps the
+`curl` command and adds the required HMAC signature and timestamp query string
+arguments. This command line tool can be used as-is to make requests to the API
+or can be embedded in a larger Python application.
+
+As additional reference, these are sections of the OTM code related to HMAC signing.
```

Comment:
It's not immediately clear to me which ACCESS_KEY credentials I'm meant to use with the HMAC tool included. Is this guide aimed at users of OTM Cloud to interact with instances and would therefore need credentials provided from the app? Perhaps a small section could be included in how to access required credentials for the signing process.

Reply:
Good question. There is not web UI for generating access tokens. It must be done through the Django shell. I added a couple paragraphs in fixup 335465dd that explain more about the generation and usage of the key pair.